Assets/Scripts/World/LettersToReid/ShadowLogic.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class ShadowLogic : Monobehavior
+{
+    public RigidBody2D playerBody;
+
+    private float shadowSpeed;
+    private float maxSpeed;
+    private int memoryCount;
+
+    void Awake()
+    {
+        //Some arbitrary max speed idk how fast a fast max speed is
+        this.maxSpeed = 10.0f;
+    }
+
+    void Update()
+    {
+        //Changes direction towards player
+        //Unsure if LookAt is appropriate for this
+        GameObject player = GameObject.Find("Player");
```

Comment:
This is a very expensive call. Should not be made every frame. You should set a reference at initialization, preferably not using `GameObject.Find`